### ISSUE TITLE
Is utf8 encoding the caller's responsibility?

### ISSUE THREAD

**Issue by grifferz** [Humanstate/business-gocardless#15]
I had a customer unable to authorise a new mandate. As soon as they went to a page that called `new_bill_url` they were getting a `Business::GoCardless::Exception=HASH(0x429caa4)` somewhere.

I tracked it down to them having an "é" in their name. I am pre-filling their first and last name in the `new_bill_url` call, so that is making its way into the JSON that is POSTed.

In Business/GoCardless/Client.pm there is:

```perl
sub _api_request {
.
.
.
.
    if ( $method =~ /POST|PUT/ ) {
        if ( $self->api_version > 1 ) {
            $req->content_type( 'application/json' );
            my $json;
            $json = JSON->new->canonical->encode( $params ) if $params;
            $req->content( $json ) if $json;
```

If I change that to:

```perl
# XXX
use Encode;
.
.
.
sub _api_request {
.
.
.
.
    if ( $method =~ /POST|PUT/ ) {
        if ( $self->api_version > 1 ) {
            # XXX
            $req->content_type( 'application/json; charset=utf8' );
            my $json;
            $json = JSON->new->canonical->encode( $params ) if $params;
            # XXX
            $req->content( encode_utf8($json) ) if $json;
```

…then GoCardless accepts it.

My knowledge of Unicode in Perl is still a bit cargo-culty. Should I be encoding the data outside of Business::GoCardless, or should Business::GoCardless be encoding it?

**Comment by leejo** [comment]
There's probably a bug here, yep. I'll do some testing on Monday.

**Comment by leejo** [comment]
@grifferz - i think you have an old version of the library. This was fixed in e556531ada6491e273f0b27d78ac7ecd06b7b443 # can you update and confirm?

**Comment by grifferz** [comment]
You're right, I was on 0.30. I'm so sorry, that is really the first thing I should have checked!
